asr1k_neutron_l3/plugins/l3/service_plugins/l3_extension_adapter.py

```diff
@@ -300,6 +300,14 @@ def get_sync_data(self, context, router_ids=None, active=None, host=None):
                     if gw_info is not None:
                         gw_info['external_fixed_ips'] = gw_port['fixed_ips']
 
+            # add address scope to internal subnets
+            all_ports = router['_interfaces'] + ([gw_port] if gw_port else [])
+            all_subnets = {sn['id'] for port in all_ports for sn in port.get('subnets', [])}
+            subnet_address_scope_map = self.db.get_subnet_address_scope_map(context, all_subnets)
+            for port in all_ports:
+                for subnet in port.get('subnets', []):
```

Comment:
you could make your code twice as fast using () instead of [] :)
The empty tuple is basically for free because it already exists and because it is immutable you always will get the same instance, while an empty list takes some time to instantiate. Not relevant here, but I find it good to know.
```
$ python -m timeit "x=()"
50000000 loops, best of 5: 7.08 nsec per loop
$ python -m timeit "x=[]"
20000000 loops, best of 5: 14.9 nsec per loop
```

Reply:
Nice idea, I'll keep it in mind. As we do it with the list in all other cases in the driver I'll keep it with the list for now.